### PR TITLE
Updated migration steps for SDN plugins

### PR DESCRIPTION
**Pull request by pravisankar**

### CODE REVIEW

**Comment by pravisankar** [comment]
Docs PR for https://github.com/openshift/origin/pull/9147

**Comment by pravisankar** [comment]
@adellape PTAL

File: install_config/configuring_sdn.adoc [hunk 1]
```diff
@@ -77,21 +77,24 @@ link:../install_config/master_node_configuration.html[node configuration file]
 ----
 networkConfig:
   mtu: 1450 <1>
-  networkPluginName: "redhat/openshift-ovs-subnet" <2>
+  networkPluginName: "" <2>
 ----
 <1> Maximum transmission unit (MTU) for the pod overlay network
-<2> Set to *redhat/openshift-ovs-subnet* for the *ovs-subnet* plug-in or
-*redhat/openshift-ovs-multitenant* for the *ovs-multitenant* plug-in
+<2> Optional for OpenShift SDN plug-in(*redhat/openshift-ovs-subnet* or *redhat/openshift-ovs-multitenant*).
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: s/OpenShift/{product-title}

File: install_config/configuring_sdn.adoc [hunk 2]
```diff
@@ -77,21 +77,24 @@ link:../install_config/master_node_configuration.html[node configuration file]
...
+Set this field if you are using third party network plug-in.
 ====
 
-[[migrating-between-sdn-plugins]]
-== Migrating Between SDN Plug-ins
+[[migrating-to-openshift-sdn-plugins]]
+== Migrating to OpenShift SDN Plug-ins
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: s/OpenShift/{product-title}

File: install_config/configuring_sdn.adoc [hunk 3]
```diff
@@ -77,21 +77,24 @@ link:../install_config/master_node_configuration.html[node configuration file]
...
 
-If you are already using one SDN plug-in and want to switch to another:
+If you are already using some SDN plug-in and want to switch to OpenShift SDN plugin:
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: s/OpenShift/{product-title}

File: install_config/configuring_sdn.adoc [hunk 4]
```diff
@@ -101,11 +104,34 @@ ifdef::openshift-enterprise[]
 *atomic-openshift-node* service on nodes.
 endif::[]
 
-When switching from the *ovs-subnet* to the *ovs-multitenant* plug-in, all the
+When switching to *ovs-multitenant* OpenShift SDN plug-in, all the
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: s/OpenShift/{product-title}

File: install_config/configuring_sdn.adoc [hunk 5]
```diff
@@ -77,21 +77,24 @@ link:../install_config/master_node_configuration.html[node configuration file]
...
-== Migrating Between SDN Plug-ins
+[[migrating-to-openshift-sdn-plugins]]
+== Migrating to OpenShift SDN Plug-ins
 
-If you are already using one SDN plug-in and want to switch to another:
+If you are already using some SDN plug-in and want to switch to OpenShift SDN plugin:
 
 . Change the `*networkPluginName*` parameter on all
-link:#configuring-the-pod-network-on-masters[masters] and
+link:#configuring-the-pod-network-on-masters[masters] in their configuration files.
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: We switched to xrefs and will only use links for external links, so:
s/link:#configuring-the-pod-network-on-masters[masters]/xref:configuring-the-pod-network-on-masters[masters]

File: install_config/configuring_sdn.adoc [hunk 6]
```diff
@@ -77,21 +77,24 @@ link:../install_config/master_node_configuration.html[node configuration file]
...
+Remove `*networkPluginName*` parameter from all
 link:#configuring-the-pod-network-on-nodes[nodes] in their configuration files.
+OpenShift node will auto detect network plugin(*ovs-subnet* or *ovs-multitenant*)
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: s/OpenShift/{product-title}

File: install_config/configuring_sdn.adoc [hunk 7]
```diff
@@ -77,21 +77,24 @@ link:../install_config/master_node_configuration.html[node configuration file]
...
+from OpenShift master.
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: s/OpenShift/{product-title}

File: install_config/configuring_sdn.adoc [hunk 8]
```diff
@@ -101,11 +104,34 @@ ifdef::openshift-enterprise[]
...
-When switching from the *ovs-subnet* to the *ovs-multitenant* plug-in, all the
+When switching to *ovs-multitenant* OpenShift SDN plug-in, all the
 existing projects in the cluster will be fully isolated (assigned unique VNIDs).
 Cluster administrators can choose to link:../admin_guide/pod_network.html[modify
 the project networks] using the administrator CLI.
 
+[[migrating-to-thirdparty-sdn-plugins]]
+== Migrating to Third Party SDN Plug-ins
+
+If you are already using OpenShift SDN plug-in and want to switch to third party plug-in:
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: s/OpenShift/{product-title}

File: install_config/configuring_sdn.adoc [hunk 9]
```diff
@@ -101,11 +104,34 @@ ifdef::openshift-enterprise[]
...
 endif::[]
 
-When switching from the *ovs-subnet* to the *ovs-multitenant* plug-in, all the
+When switching to *ovs-multitenant* OpenShift SDN plug-in, all the
 existing projects in the cluster will be fully isolated (assigned unique VNIDs).
 Cluster administrators can choose to link:../admin_guide/pod_network.html[modify
 the project networks] using the administrator CLI.
 
+[[migrating-to-thirdparty-sdn-plugins]]
+== Migrating to Third Party SDN Plug-ins
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: s/Third Party/Third-party

File: install_config/configuring_sdn.adoc [hunk 10]
```diff
@@ -101,11 +104,34 @@ ifdef::openshift-enterprise[]
...
+link:#configuring-the-pod-network-on-nodes[nodes] in their configuration files.
+ifdef::openshift-origin[]
+. Restart the *origin-master* service on masters and the *origin-node* service
+on nodes.
+endif::[]
+ifdef::openshift-enterprise[]
+. Restart the *atomic-openshift-master* service on masters and the
+*atomic-openshift-node* service on nodes.
+endif::[]
+. Cleanup OpenShift SDN specific artifacts:
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: s/OpenShift/{product-title}

File: install_config/configuring_sdn.adoc [hunk 11]
```diff
@@ -101,11 +104,34 @@ ifdef::openshift-enterprise[]
...
 Cluster administrators can choose to link:../admin_guide/pod_network.html[modify
 the project networks] using the administrator CLI.
 
+[[migrating-to-thirdparty-sdn-plugins]]
+== Migrating to Third Party SDN Plug-ins
+
+If you are already using OpenShift SDN plug-in and want to switch to third party plug-in:
+
+. Change the `*networkPluginName*` parameter on all
+link:#configuring-the-pod-network-on-masters[masters] and
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: s/link:#configuring-the-pod-network-on-masters[masters]/xref:configuring-the-pod-network-on-masters[masters]
s/link:#configuring-the-pod-network-on-nodes[nodes]/xref:configuring-the-pod-network-on-nodes[nodes]

File: install_config/configuring_sdn.adoc [hunk 12]
```diff
@@ -77,21 +77,24 @@ link:../install_config/master_node_configuration.html[node configuration file]
...
 networkConfig:
   mtu: 1450 <1>
-  networkPluginName: "redhat/openshift-ovs-subnet" <2>
+  networkPluginName: "" <2>
 ----
 <1> Maximum transmission unit (MTU) for the pod overlay network
-<2> Set to *redhat/openshift-ovs-subnet* for the *ovs-subnet* plug-in or
-*redhat/openshift-ovs-multitenant* for the *ovs-multitenant* plug-in
+<2> Optional for OpenShift SDN plug-in(*redhat/openshift-ovs-subnet* or *redhat/openshift-ovs-multitenant*).
+Set this field if you are using third party network plug-in.
```
Comment (ahardin-rh) [inline install_config/configuring_sdn.adoc]: s/third party/third-party

**Comment by ahardin-rh** [comment]
@pravisankar Thanks! Just a few small comments from me. This also needs a rebase.

**Comment by pravisankar** [comment]
@ahardin-rh Updated, please review and merge.

**Comment by ahardin-rh** [comment]
@pravisankar Thanks!

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../install_config/configuring_sdn.adoc#install-config-configuring-sdn[Configuring the SDN]
|Updated migration steps for SDN plug-ins.
%

**Pull request merged**